cmake/update_pot_source_dependencies.cmake

```diff
@@ -30,13 +30,22 @@ if(DOMAIN STREQUAL ${DEFAULT_DOMAIN})
 				while read file\; do
 					# If the file doesn't contain a GETTEXT_DOMAIN
 					# definition it should be added to the default domain.
-					if ! grep \"^\#define  *GETTEXT_DOMAIN\"
+					if ! grep '^\#define  *GETTEXT_DOMAIN'
 							$$file > /dev/null 2>&1\; then
 
-						echo $$file >>
-							${PROJECT_SOURCE_DIR}/po/${DOMAIN}/POTFILES.in \;
+						echo $$file \;
+
+					# While files don't need a GETTEXT_DOMAIN to be included in
+					# the default domain, accept files that contain a
+					# GETTEXT_DOMAIN definition for the default domain too.
+					elif grep '^\#define  *GETTEXT_DOMAIN  *\"${DOMAIN}\"'
+							$$file > /dev/null 2>&1\; then
+
+						echo $$file \;
 					fi
-				done
+				done >>
```

Comment:
Presumably it can be a non-append redirection now? Or is the file expected to be there already?

Reply:
True, it could be. The file's created by a `touch` command in this cmake file.

I'm not sure if we'd need to worry about Bash's `noclobber` option, which makes it an error to `>` to an already-existing file.

Reply:
Right, just before this command. Since now there is no chance the file won't exist that is probably useless.

Based on the touch command `noclobber` is clearly not on so far and this should not read the user's .bashrc or so either.

Reply:
I don't understand. Why does the touch command tell us anything about `noclobber`?

Reply:
Sorry, I misunderstood. I'd be surprised if `noclobber` is enabled by default anywhere. That'd break lots of scripts. If you're really worried you could make it `>|`.

Reply:
Changed it to that, and removed the `touch` lines.